bindings/go/src/ucx/request.go

```diff
@@ -83,15 +91,15 @@ func isRequestPtr(request C.ucs_status_ptr_t) bool {
 	return (uint64(uintptr(request)) - 1) < (uint64(errLast) - 1)
 }
 
-func NewRequest(request C.ucs_status_ptr_t, callbackId uint64, immidiateInfo interface{}) (*UcpRequest, error) {
+func NewRequest(request C.ucs_status_ptr_t, callback UcpCallback, immidiateInfo interface{}) (*UcpRequest, error) {
 	ucpRequest := &UcpRequest{}
 
 	if isRequestPtr(request) {
 		ucpRequest.request = unsafe.Pointer(uintptr(request))
 		ucpRequest.Status = UCS_INPROGRESS
 	} else {
 		ucpRequest.Status = UcsStatus(int64(uintptr(request)))
-		if callback, found := deregister(callbackId); found {
+		if callback != nil {
```

Comment:
looks like leak, in this case handle was created and never destroyed

Reply:
Maybe I'm missing something, if `callback` is nil I don't pack it. Check `packParams`